app/models/requirement.rb

```diff
@@ -81,20 +87,31 @@ def has_data_validation?
 
   private
 
+  def using_defaulted_requirement_code
+    uuid_regex = /^dummy-[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i
+    uuid_regex.match?(self.requirement_code)
+  end
+
   # requirement codes should not be auto generated during seeding.  Use uuid if not provided
   def set_requirement_code
-    self.requirement_code ||=
-      (
-        if label.present?
-          label.parameterize.underscore.camelize(:lower)
-        else
-          SecureRandom.uuid
-        end
-      )
+    using_default = self.using_defaulted_requirement_code
+    blank = self.requirement_code.blank?
+
+    return unless using_default || blank
 
-    return unless input_type == "file" && !requirement_code.end_with?("_file")
+    needs_file_suffix = input_type == "file" && !requirement_code.end_with?("_file")
```

Comment:
this should be before blank and use_default check. This is because file input's must have _file at the end, as sunny uses it for some file upload related logic I think